src/nova_deno.ts

```diff
@@ -70,6 +77,69 @@ function watchConfigFile(workspacePath: string, filename: string) {
   compositeDisposable.add(configWatcher);
 }
 
+/**
+ * This is used to determine whether certain features, like the "Find Symbol"
+ * command, are available.
+ */
+function registerEditorWatcher() {
+  // This callback is called immediately and once. Then, it is called every time the user opens a text editor.
+  const disposable = new CompositeDisposable();
+
+  const compatibleSyntaxes = ["typescript", "tsx", "javascript", "jsx"];
+  function checkFeatureAvailability(
+    documents: readonly TextDocument[],
+  ) {
+    const syntaxes = documents.map((document) => document.syntax ?? "plain");
+    const isCompatible = compatibleSyntaxes.some((syntax) =>
+      syntaxes.includes(syntax)
+    );
+
+    if (isCompatible) {
+      // @ts-expect-error: The Nova types are outdated. This feature was added in version 5.
+      (nova.workspace.context as Configuration).set(
+        "shouldDisplayFeatures",
+        true,
+      );
+    } else {
+      // @ts-expect-error: for the reason above
+      (nova.workspace.context as Configuration).set(
+        "shouldDisplayFeatures",
+        false,
+      );
+    }
+  }
+
+  disposable.add(
+    nova.workspace.onDidAddTextEditor((editor) => {
+      checkFeatureAvailability(nova.workspace.textDocuments);
+
+      disposable.add(editor.document.onDidChangeSyntax(() => {
+        checkFeatureAvailability(nova.workspace.textDocuments);
+      }));
+
+      disposable.add(editor.onDidDestroy((deletedEditor) => {
+        const editors: (TextEditor | null)[] = [...nova.workspace.textEditors];
```

Comment:
Huh. I would have thought that Nova would remove the editors from the list of text editors after it's destroyed. But maybe it's more useful for us that this isn't the case.

Reply:
I assumed it'd do that too. Instead, the array of text editors seems to include the deleted editor by the time the callback runs.